src/main/java/uk/gov/register/util/CanonicalJsonValidator.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.register.util;
+
+import uk.gov.register.exceptions.SerializationFormatValidationException;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+public class CanonicalJsonValidator {
+    private final JsonMapper jsonMapper;
+    private final CanonicalJsonMapper canonicalJsonMapper;
+
+    public CanonicalJsonValidator() {
+        this.jsonMapper = new JsonMapper();
+        this.canonicalJsonMapper = new CanonicalJsonMapper();
+    }
+
+    public void validateItemStringIsCanonicalized(String jsonItem) {
+        byte[] jsonItemBytes = jsonItem.getBytes(StandardCharsets.UTF_8);
+        byte[] nonCanonicalizedBytes = jsonMapper.writeToBytes(jsonMapper.readFromBytes(jsonItemBytes));
+        byte[] canonicalizedBytes = canonicalJsonMapper.writeToBytes(canonicalJsonMapper.readFromBytes(jsonItemBytes));
+
+        if (!Arrays.equals(nonCanonicalizedBytes, canonicalizedBytes)) {
```

Comment:
We should be comparing jsonItemBytes to canonicalizedBytes.

nonCanonicalizedBytes in this case will have removed any white space that appeared in jsonItemBytes and therefore prevents us from detecting white space in jsonItemBytes. Therefore nonCanonicalizedBytes should be removed and not used.